src/ec/packages/ec-component-site-header-harmonised/site-header-harmonised.story.js

```diff
@@ -41,44 +55,118 @@ const btnLoginHandler = () => {
   }
 };
 
+const prepareSiteHeaderHarmonised = data => {
+  data.group = select('group', [data.group], data.group, buttonLabels.required);
+  if (data.banner_top && data.banner_top.link) {
+    data.banner_top.link.label = text(
+      'banner_top.link.label',
+      data.banner_top.link.label,
+      buttonLabels.required
+    );
+    data.banner_top.link.path = text(
+      'banner_top.link.path',
+      data.banner_top.link.path,
+      buttonLabels.required
+    );
+  }
+  data.icon_file_path = select(
+    'icon_file_path',
+    [defaultSprite],
+    defaultSprite,
+    buttonLabels.required
+  );
+  let logoDefault = logo;
+  if (data.logo.src !== '/logo--en.svg') {
+    logoDefault = data.logo.src;
+  }
+  data.logo.src = select(
+    'logo.src',
+    [logoDefault],
+    logoDefault,
+    buttonLabels.required
+  );
+  // Logo knobs
+  getLogoKnobs(data);
+  // Login box and login toggle knobs.
+  if (data.login_box) {
+    data.logged = boolean('logged', data.logged, buttonLabels.states);
+    getLoginKnobs(data);
+  }
+  // Search toggle.
+  if (data.search_toggle && data.search_toggle.label) {
+    data.search_toggle.label = text(
+      'search_toggle.label',
+      data.search_toggle.label,
+      buttonLabels.required
+    );
+    data.search_toggle.href = text(
+      'search_toggle.href',
+      data.search_toggle.href,
+      buttonLabels.required
+    );
+  }
+  // Search form.
+  if (data.search_form) {
+    getSearchFormKnobs(data);
+  }
+  // Language selector knobs.
+  if (data.language_selector) {
+    getLanguageSelectorKnobs(data, false);
+    // Language selector overlay.
+    data.language_selector.overlay = object(
+      'language_selector.overlay',
+      data.language_selector.overlay,
+      buttonLabels.required
+    );
+  }
+  // Extra classes and extra attributes.
+  getExtraKnobs(data);
+  if (data.menu) {
+    // Menu label.
+    data.menu_label = text(
+      'data.menu_label',
+      data.menu_label,
+      buttonLabels.optional
+    );
+    data.menu = object('data.menu', data.menu, buttonLabels.optional);
+  }
+
+  return data;
+};
 storiesOf('Components/Site Headers/Harmonised', module)
   .addDecorator(withNotes)
   .addDecorator(withCode)
   .addDecorator(withKnobs)
   .add(
     'group 1',
```

Comment:
Maybe a nice use case to have is:
`with or without menu`
What do you think?

Regarding this I think we have a problem here(more on ecl side), as the menu is optional and the site name mandatory, if I remove the menu, there is something wrong with padding margin.. also the site-name is a child  element of the nav tag. shouldn't it be separated from navigation?

Reply:
What the heck...good that you noticed this, not easy to spot it, the only example if i'm not mistaken is in that "event demo" on ECL.
It wasn't too hard to adapt the template, it's more the stories that changed a lot, there is almost  no required element anymore.